#172019567 setup Redux for the whole app

hakoemmy opened the pull request:
#### What does this PR do?
- This PR setups up Redux
#### Description of Task to be completed?
- Implement boilerplate real-life example that doesn't add value to the LMS to setup and demonstrate the redux flow
- add ```redux``` module
- add ```react-redux``` to read updates from the redux store
- add ```redux-devtools-extension``` for debugging application state changes in the browser
- add ```redux-logge``` middleware to log store in the browser console
- add ```redux-thunker``` middleware to enable action creators to return a function instead of action object
#### How should this be manually tested?
- Clone the repo via ```git clone <repo-link>```
- Run ```npm install``` to install all necessary dependencies
- Run ```npm start``` to start the server locally
- Enable Redux devtools extension in your chrome browser, here is the attached [link](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) to do so. This will help you to track application state store in the browser
- Inspect the browser to see app state logs created by ```redux-logger``` middleware on fly
- Then test the whole functionality like so. Above all take a look at below screenshots to have a clue on how I tested everything.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#172019567](https://www.pivotaltracker.com/story/show/172019567)
### Screenshots (if appropriate)
<img width="1135" alt="Screen Shot 2020-03-29 at 19 14 40" src="https://user-images.githubusercontent.com/51243858/77855586-950dfb80-71f1-11ea-8d1a-ee835e9bf5c0.png">
<img width="797" alt="Screen Shot 2020-03-29 at 19 14 15" src="https://user-images.githubusercontent.com/51243858/77855593-9f2ffa00-71f1-11ea-9e5e-49fda69d080c.png">
<img width="776" alt="Screen Shot 2020-03-29 at 19 13 08" src="https://user-images.githubusercontent.com/51243858/77855600-af47d980-71f1-11ea-865f-cf87867cc113.png">
<img width="795" alt="Screen Shot 2020-03-29 at 19 12 46" src="https://user-images.githubusercontent.com/51243858/77855609-bcfd5f00-71f1-11ea-9de1-ecfc3ff35b28.png">
#### Questions:
N/A

hakoemmy committed:
chore(redux): setup redux for the whole app

- implement boilerplate reallife example to demonstrate redux flow
- add redux module
- add react-redux to read updates from redux store
- add redux-devtools-extension for debugging application state changes in browser
- add redux-logger middlware to log store in browser console
- add redux-thunker middlware to enable action creators return a function instead of action object

[Maintains #172019567]

blaise82 commented:
- please don't write CSS outside of a styling folder
-  from my point of view, I think it would have been better if you wrote tests for it 
-  please if you change somethings in files make sure that tests are running as they are supposed to not failing
-  the branch name missed something (type)
- just a suggestion if possible can this PR just be there for setting up  redux not writing any other things 

- like for example, the only new files we should be having are 

1. src/store/index.js
2. src/store/reducers/index.js

- so that we may learn things slowly by slowly.
- PLEASE REBASE
 **thanks**

hakoemmy commented:
> * please don't write CSS outside of a styling folder
> * from my point of view, I think it would have been better if you wrote tests for it
> * please if you change somethings in files make sure that tests are running as they are supposed to not failing
> * the branch name missed something (type)
> * just a suggestion if possible can this PR just be there for setting up  redux not writing any other things
> * like for example, the only new files we should be having are
> 
> 1. src/store/index.js
> 2. src/store/reducers/index.js
> 
> * so that we may learn things slowly by slowly.
> * PLEASE REBASE
>   **thanks**

> * please don't write CSS outside of a styling folder
>> I used CSS modules, that means a CSS file resides in the same folder with its component
> * from my point of view, I think it would have been better if you wrote tests for it
>> For sure, I can do this but since the example, I've used to demonstrate doesn't add value to the project. That means we'll even remove some functionality of it once the first feature is implemented. I decided to leave unit testing behind but we'll be testing Redux and react components for sure.
> * please if you change somethings in files make sure that tests are running as they are supposed to not failing
>> Sure thing, I've seen that you intentionally merged a PR for``` setting up travis``` after I raised mine so that's why the tests were failing but I've fixed them like so
> * the branch name missed something (type)
>> Ooh, Thanks for this. I forgot to it but I've renamed the branch-like so
> * just a suggestion if possible can this PR just be there for setting up  redux not writing any other things
>> My friend, I've even explained in the PR description that I've used a life example to demonstrate Redux flow. That means the example of counter used will be removed very soon like I said.
>* like for example, the only new files we should be having are
> 
>1. src/store/index.js
> 2. src/store/reducers/index.js
> 
 >* so that we may learn things slowly by slowly.
>>> I understand your suggestions of having   these two files: 
```
>1. src/store/index.js
> 2. src/store/reducers/index.js
``` 
>>> But I needed to demonstrate how Redux works with an actual real-life example. and for sure this would be very easier for us to quickly integrate other features in the Redux store I created
